app/controllers/users/forget_all_devices_controller.rb

```diff
@@ -0,0 +1,18 @@
+module Users
+  class ForgetAllDevicesController < ApplicationController
+    before_action :authenticate_user!
```

Comment:
Doesn't `confirm_two_factor_authenticated` call this?

Reply:
Honestly I didn't check, it was just some copy-pasta from another controller. I will dig in and confirm

Reply:
Looked it up. Here's where it is called in `confirm_two_factor_authenticated`: https://github.com/18F/identity-idp/blob/master/app/controllers/application_controller.rb#L216

Reply:
🙈 thanks! fixing